### ISSUE TITLE
Hugging Face Support

### ISSUE THREAD

**Issue by GustyCube** [GustyCube/EasilyAI#17]
We need to implement a bunch of hugging face features, which will be a lot of work. We have:

- [ ] Multimodal
- [ ] Audio-Text-to-Text
- [ ] Image-Text-to-Text
- [ ] Visual Question Answering
- [ ] Document Question Answering
- [ ] Video-Text-to-Text
- [ ] Any-to-Any
- [ ] Computer Vision
- [ ] Depth Estimation
- [ ] Image Classification
- [ ] Object Detection
- [ ] Image Segmentation
- [ ] Text-to-Image
- [ ] Image-to-Text
- [ ] Image-to-Image
- [ ] Image-to-Video
- [ ] Unconditional Image Generation
- [ ] Video Classification
- [ ] Text-to-Video
- [ ] Zero-Shot Image Classification
- [ ] Mask Generation
- [ ] Zero-Shot Object Detection
- [ ] Text-to-3D
- [ ] Image-to-3D
- [ ] Image Feature Extraction
- [ ] Keypoint Detection
- [ ] Natural Language Processing
- [ ] Text Classification
- [ ] Token Classification
- [ ] Table Question Answering
- [ ] Question Answering
- [ ] Zero-Shot Classification
- [ ] Translation
- [ ] Summarization
- [ ] Feature Extraction
- [ ] Text Generation
- [ ] Text2Text Generation
- [ ] Fill-Mask
- [ ] Sentence Similarity
- [ ] Audio
- [ ] Text-to-Speech
- [ ] Text-to-Audio
- [ ] Automatic Speech Recognition
- [ ] Audio-to-Audio
- [ ] Audio Classification
- [ ] Voice Activity Detection
- [ ] Tabular
- [ ] Tabular Classification
- [ ] Tabular Regression
- [ ] Time Series Forecasting
- [ ] Reinforcement Learning
- [ ] Reinforcement Learning
- [ ] Robotics
- [ ] Other
- [ ] Graph Machine Learning

And their libraries: 

- [ ] PyTorch
- [ ] TensorFlow
- [ ] JAX
- [ ] Safetensors
- [ ] Transformers
- [ ] PEFT
- [ ] TensorBoard
- [ ] GGUF
- [ ] Diffusers
- [ ] stable-baselines3
- [ ] ONNX
- [ ] sentence-transformers
- [ ] ml-agents
- [ ] TF-Keras
- [ ] Adapters
- [ ] setfit
- [ ] MLX
- [ ] timm
- [ ] sample-factory
- [ ] Keras
- [ ] Flair
- [ ] Transformers.js
- [ ] OpenVINO
- [ ] spaCy
- [ ] fastai
- [ ] ESPnet
- [ ] Joblib
- [ ] NeMo
- [ ] Core ML
- [ ] BERTopic
- [ ] TF Lite
- [ ] OpenCLIP
- [ ] Rust
- [ ] fastText
- [ ] Scikit-learn
- [ ] KerasHub
- [ ] speechbrain
- [ ] PaddlePaddle
- [ ] Fairseq
- [ ] Asteroid
- [ ] llamafile
- [ ] AllenNLP
- [ ] Graphcore
- [ ] Stanza
- [ ] paddlenlp
- [ ] SpanMarker
- [ ] Habana
- [ ] pyannote.audio
- [ ] unity-se
- [ ] ntis
- [ ] DDUF

And different languages :)

**Comment by sgarg26** [comment]
Hey, this looks interesting. Could you elaborate a little bit on how you would like this to be implemented? I'm interested in implementing the depth estimation functionality because I've used some of the models on hugging face. Thanks

**Comment by GustyCube** [comment]
Hey @sgarg26! Sorry for the wait. I'd like this implemented where each hugging face functionality would hook into a main hugging face service, under categories, and libraries probably need a separate service that can fit in with HuggingFace.

**Comment by GustyCube** [comment]
@sgarg26 does this make sense?

**Comment by sgarg26** [comment]
@GustyCube Sorry, I've been busy, I completely missed this. If I'm understanding what you're saying correctly, then doesn't Transformers offer a similar service already? 

It would be interesting to have an API that lets you run these models without having to download them, which is what Transformers does, but that would require actually hosting them somewhere. 

Thoughts?

**Comment by GustyCube** [comment]
@sgarg26 I think it depends on your use case... remember hugging face itself encompasses many different libraries, and I'm trying to incorporate all of them.

**Comment by sgarg26** [comment]
@GustyCube I see. I think this might be a little too complex for me to do at this time unfortunately, but I'd love to revisit it later :)
